plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/utils/JsonElementToLongSerializerDeserializerTest.kt

```diff
@@ -0,0 +1,71 @@
+package org.wordpress.android.fluxc.utils
+
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonPrimitive
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class JsonElementToLongSerializerDeserializerTest {
+    private val sut = JsonElementToLongSerializerDeserializer()
+
+    @Test
+    fun `when value is true then null value is returned`() {
+        val jsonElement = JsonPrimitive(true)
+        val result = sut.deserialize(
+            json = jsonElement,
+            typeOfT = null,
+            context = null
+        )
+        assertThat(result).isNull()
+    }
+
+    @Test
+    fun `when value is a valid string number then the value is returned`() {
+        val value = 12345L
+        val valueString = value.toString()
+        val jsonElement = JsonPrimitive(valueString)
+        val result = sut.deserialize(
+            json = jsonElement,
+            typeOfT = null,
+            context = null
+        )
+        assertThat(result).isEqualTo(value)
+    }
+
+    @Test
+    fun `when value is an invalid string number then the value is returned`() {
+        val jsonElement = JsonPrimitive("12345_test")
+        val result = sut.deserialize(
+            json = jsonElement,
+            typeOfT = null,
+            context = null
+        )
+        assertThat(result).isNull()
+    }
+
+    @Test
+    fun `when value is different number format then the value is parsed to long and returned`() {
+        val value = 12345L
+        val valueInt = value.toDouble()
+        val jsonElement = JsonPrimitive(valueInt)
+        val result = sut.deserialize(
+            json = jsonElement,
+            typeOfT = null,
+            context = null
+        )
+        assertThat(result).isEqualTo(value)
+    }
+
+    @Test
+    fun `test serialization deserialization`() {
+        val gsonBuilder = GsonBuilder()
+        val gson = gsonBuilder.registerTypeAdapter(Long::class.java, sut).create()
```

Comment:
What's the reason for registering Long adapter? I think it's not necessary 
```suggestion
        val gson = Gson()
```